### ISSUE TITLE
Node wrangler addon improvements

### ISSUE THREAD

**Issue by ReinerBforartists** [Bforartists/Bforartists#3075]
Yes i know, we should not maintain addons. But i tried to use the Node Wrangler Addon . And failed miserably at it. How the heck do you add a viewer node? The hotkey shift + ctrl + lmb for it seems to do nothing. Also not in Blender.

Nearly the whole functionality of the Node Wrangler addon is hotkey only. There is a right click menu, but even here, nothing is to discover really. There are no tooltips. And also here, the viewer node is missing. So it would be a good idea to have a menu in the header with all its functionality ...

Basically it would already help to append the existing node wrangler menu in the header. We have this space. And add the missing bits then.

![nodewranglermenu](https://user-images.githubusercontent.com/13919702/174757904-41363ff0-079c-407c-8c5e-b62c3d2173db.jpg)

**Comment by Blenux** [comment]
Strange that keymap works for me?, also the Viewer is just a renamed Colored Emission Shader, and the preferences has the long list of hotkeys, though I only use the viewer one. 

https://user-images.githubusercontent.com/25260650/174791881-ac210a01-9d04-49ff-829f-0c783697b9e8.mp4

.

**Comment by Draise14** [comment]
This addon also has a tab and panel to the side in the nodeshelf, which contains most operators there. It might not need to be in the header. Though having some of these operators in our own toolshelf would be handy.

**Comment by ReinerBforartists** [comment]
Okay, i haven't realized that the content is already in the sidebar. What we still could do is to display the list of hotkeys in the sidebar in a panel.

To add some of the operators to the toolbar is no good idea though. The addon is not enabled by default. And we have no node editor related toolbar active here. Or do you mean the tool shelf? Name confusion. Blender calls the old tool shelf now toolbar. While we still have both ...

**Comment by Blenux** [comment]
It also adds these in the Shader Editor - Add - Texture menu with no Icons.

I use the Multiple Images all the time.

![Screenshot 2022-06-22 181837](https://user-images.githubusercontent.com/25260650/174979919-c1e91525-4e4d-4972-b77c-6dd17eb17eb6.png)

**Comment by Draise14** [comment]
Hmm.... initial gut feeling is to not maintain these entries from the addon, but we DO have this enabled as a default, so better integration would be consistent.

**Comment by Draise14** [comment]
Just to clarify, what really needs to be improved?

**Current Features:**
- Custom Hotkeys and features
- Property Shelf panels with menus and buttons
- Custom context menu
- Adds two entries to the Texture Add menu (no icons)

What could we immediately improve?
- Add icons to the Texture Add Menu entries
- Add and expose the two image entries to the Add Tab of the Nodeshelf

Pro: This could be an easy maintenance and quick to implement. 
Con: If the addon is disabled, the Add Tab in the nodeshelf may bug since those are not registered? 

Future improvement?:
- Make a tab and expose some common buttons to the Tool Shelf?

**Comment by ReinerBforartists** [comment]
> Con: If the addon is disabled, the Add Tab in the nodeshelf may bug since those are not registered?

These entries would have to come from the addon then. 

The question still remains if we should touch the addon at all. What i had in mind was one or two small changes. What we have now is half a rewrite of the addon :D

**Comment by Draise14** [comment]
I think let's just let it be as a policy. Maybe in the future when we have more hands on deck, then it would make sense to maintain third party addons and include some - but for now no. So will close.
